fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/ReindexManager.java

```diff
@@ -116,7 +116,11 @@ public void start() throws InterruptedException {
             for (final var worker : workers) {
                 worker.join();
             }
-            indexMembership();
+            if (failOnError && errorCount.get() == 0) {
```

Comment:
Not quite sure I follow this conditional, why would we only `indexMembership` when `failOnError` is set to true?

Reply:
This was an additional bug that was discovered by UVA. By default, when the indexing fails, it should short-circuit. However, what was actually happening was that when the indexing failed, it would short-circuit the object indexing but still attempt to index membership. This new conditional was intended to fix that problem so that membership would not be indexed if it's supposed to fail on error and there are errors.

That said, there's a bug in the new logic as it will not index membership when `failOnError` is `false`. Will fix.